### PR TITLE
[UNR-3680][MS] Spatial connection manager created from spatial net driver

### PR DESCRIPTION
**Pull request by MatthewSandfordImprobable**
If the AutoPreventCloudConnection setting is enabled on clients it is possible that the first connection will be attempted without a connection manager if bPersistSpatialConnection is true. We now attempt to "StartSpatalConnection" from the netdriver which itself will create a connection manager and inject the a locator into the command line arguments.

### CODE REVIEW

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp [hunk 1]
```diff
@@ -222,7 +222,12 @@ void USpatialNetDriver::InitiateConnectionToSpatialOS(const FURL& URL)
 		bPersistSpatialConnection = URL.HasOption(*SpatialConstants::ClientsStayConnectedURLOption);
 	}
 
-	if (!bPersistSpatialConnection)
+	// This will create a Spatial Connection Manager if it is our first time connection.
+	// It will also inject a command line locater.
+	bool bCreatedNewNetDriver = GameInstance->TryStartSpatialConnection();
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:227]: Now that this is here, couldn't it be removed from all other places?

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameInstance.cpp [hunk 2]
```diff
@@ -114,18 +114,25 @@ FGameInstancePIEResult USpatialGameInstance::StartPlayInEditorGameInstance(ULoca
...
 		// If we are using spatial networking then prepare a spatial connection.
-		TryInjectSpatialLocatorIntoCommandLine();
 		CreateNewSpatialConnectionManager();
+
+		bHasPreviouslyConnectedToSpatial = true;
+
+		return true;
 	}
 #if TRACE_LIB_ACTIVE
 	else
```
Comment (m-samiec) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialGameInstance.cpp:138]: We don't want this code running in the else case

File: SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp (hunk 1 above)
Comment (MatthewSandfordImprobable) [inline SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:227]: Should read bCreatedNewSpatialConnectionManager

Comment (MatthewSandfordImprobable) [reply SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp:227]: Yes you might be right.

**Pull request closed without merging**